### PR TITLE
Autodetect ndk path

### PR DESCRIPTION
**Pull request by Tormund**
Hardcoded paths removed, now we search in %PATH% ($PATH) or in envvars,
for SDL we need to set SDL_HOME envvar.
Now nimParallelBuild set NDK '-j' arg

### CODE REVIEW

**Comment by Tormund** [comment]
I think we can test this on Unix systems.

File: nimx/naketools.nim [hunk 1]
```diff
@@ -79,6 +79,17 @@ proc setBuilderSettings(b: Builder) =
...
+proc replaceInStr(in_str, wh_str : string, by_str: string = ""): string =
+    result = in_str
+    if in_str.len > 0:
+        var pos = in_str.rfind(wh_str)
+        result.delete(pos, result.len)
+        if by_str.len > 0:
+            result &= by_str
+
+proc getEnvErrorMsg(env: string): string =
+    result = "\nEnviroment [" & env & "] not exist, verify your settings!"
```
Comment (yglukhov) [inline nimx/naketools.nim]: Environment variable BLA_BLA is not set.

File: nimx/naketools.nim [hunk 2]
```diff
@@ -89,17 +100,58 @@ proc newBuilder(platform: string): Builder =
     b.javaPackageId = "com.mycompany.MyGame"
     b.disableClosureCompiler = false
 
+    ## try find binary for android sdk, ndk, and nim
```
Comment (yglukhov) [inline nimx/naketools.nim]: Please do this only for android target

File: nimx/naketools.nim [hunk 3]
```diff
@@ -89,17 +100,66 @@ proc newBuilder(platform: string): Builder =
...
+        if sdk_path.isNil: error_msg &= getEnvErrorMsg("ANDROID_HOME")
+        if ndk_path.isNil: error_msg &= getEnvErrorMsg("NDK_HOME")
+        if nim_path.isNil: error_msg &= getEnvErrorMsg("NIM_HOME")
+
+        b.androidSdk = sdk_path
+        b.androidNdk = ndk_path
+        b.nimIncludeDir = nim_path
+
+    var sdl_home : string
+    if existsEnv("SDL_HOME"):
```
Comment (yglukhov) [inline nimx/naketools.nim]: This should also be done only for ios/android

**Pull request merged**